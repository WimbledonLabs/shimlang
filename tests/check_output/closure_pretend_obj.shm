fn Obj() {
    fn foo() {
        print("foo'd");
    }

    fn bar() {
        print("bar'd");
    }

    fn run_fn(method) {
        if method == "foo" {
            foo();
        } else if method == "bar" {
            bar();
        } else {
            print("unknown fn", method);
        }
    }

    return run_fn;
}

let obj = Obj();
obj("foo");
obj("bar");
obj("can");
