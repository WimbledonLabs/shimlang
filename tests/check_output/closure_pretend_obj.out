foo'd
bar'd
unknown fn can
